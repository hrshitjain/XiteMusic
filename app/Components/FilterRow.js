import React  from 'react';
import {View, Text} from 'react-native'
import CheckBox from '@react-native-community/checkbox';
import { filterRowStyles } from '../Styles/PopupStyles';

const FilterRow = ({item, onValueChange, selectedValues}) =>{
    const isSelected = selectedValues.indexOf(item.id) > -1;
    return(
        <View style={filterRowStyles.rowItem}>
            <CheckBox style={filterRowStyles.checkbox}
                      testID="Filter.Check"
                      value={isSelected}
                      onChange={() => onValueChange(item.id)}
                      disabled={false}/>
            <Text style={filterRowStyles.title}>{item.name}</Text>
        </View>
    )
}

export default FilterRow;