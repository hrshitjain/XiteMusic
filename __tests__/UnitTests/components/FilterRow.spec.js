import React from 'react';
import { render } from '@testing-library/react-native';
import FilterRow from '../../../app/Components/FilterRow';

const defaultProps = {
    item: {
        name:"Pop",
        id:1
    },
    onValueChange: jest.fn(),
    selectedValues:[]
}

describe('FilterRow component render', () =>{

    it('Snapshot test', () =>{
        const {toJSON} = render(<FilterRow {...defaultProps}/>);
        expect(toJSON()).toMatchSnapshot();
    });
    it('Root View has 2 children', () => {
        const {toJSON} = render(<FilterRow {...defaultProps}/>);
        expect(toJSON().children.length).toBe(2);
    });
    it('Genre Label is rendered',()=>{
        const {getAllByText} = render(<FilterRow {...defaultProps}/>);
        expect(getAllByText(defaultProps.item.name).length).toBe(1);
    });
    it('Genre Checkbox is rendered',()=>{
        const {getByTestId} = render(<FilterRow {...defaultProps}/>);
        expect(getByTestId("Filter.Check").length).toBe(1);
    });
});