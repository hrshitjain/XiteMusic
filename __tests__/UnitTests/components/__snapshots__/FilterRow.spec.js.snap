// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterRow component render Snapshot test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "padding": 10,
    }
  }
>
  <View
    accessibilityRole="checkbox"
    accessibilityState={
      Object {
        "checked": false,
        "disabled": false,
      }
    }
    accessibilityValue={
      Object {
        "text": "off",
      }
    }
    accessible={true}
    pointerEvents="auto"
    testID="Filter.Check"
  >
    <RNCCheckbox
      forwardedRef={null}
      onChange={[Function]}
      onValueChange={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "height": 32,
            "width": 32,
          },
          Object {
            "margin": 10,
          },
        ]
      }
      value={false}
    />
  </View>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 14,
        "fontWeight": "bold",
        "marginLeft": 10,
      }
    }
  >
    Pop
  </Text>
</View>
`;
